src/main.cpp

```diff
@@ -4617,26 +4617,31 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
     else if (strCommand == "mempool")
     {
-        LOCK2(cs_main, pfrom->cs_filter);
 
-        std::vector<uint256> vtxid;
-        mempool.queryHashes(vtxid);
-        vector<CInv> vInv;
-        BOOST_FOREACH(uint256& hash, vtxid) {
-            CInv inv(MSG_TX, hash);
-            CTransaction tx;
-            bool fInMemPool = mempool.lookup(hash, tx);
-            if (!fInMemPool) continue; // another thread removed since queryHashes, maybe...
-            if ((pfrom->pfilter && pfrom->pfilter->IsRelevantAndUpdate(tx)) ||
-               (!pfrom->pfilter))
-                vInv.push_back(inv);
-            if (vInv.size() == MAX_INV_SZ) {
-                pfrom->PushMessage("inv", vInv);
-                vInv.clear();
+        //Put the mempool on a 16 to 32 second quantized time delay to suppress the arrival time information leak.
+        int64_t rounded_time = (GetTime() - 16) & 0x7ffffffffffffff0LL;
```

Comment:
I can see this confusing someone in the future (I had to stop and think about it).

Needs a comment like "equivalent to (GetTime() - 16) % 16 but faster"